Extract Toggle from Checkbox

jason0x43 opened the pull request:
Toggle is a separate component that extends Checkbox.

references #455

**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

**Description:**

This PR splits Toggle out of Checkbox.

Resolves #455

tomdye commented on src/checkbox/Checkbox.ts:
```diff
@@ -98,29 +82,7 @@ export class CheckboxBase<P extends CheckboxProperties = CheckboxProperties> ext
 	}
 
 	protected renderToggle(): DNode[] {
```
`renderToggle` can be removed from here

jason0x43 replied:
It's necessary so it can be called in `render` and overridden in Toggle (which keeps Toggle from having to duplicate all the render code).

tomdye commented:
Also, don't forget to raise a change in `@dojo/themes` when this is complete

jason0x43 commented:
I reworked this a bit to better share functionality between Checkbox and Toggle. Almost everything is shared; the only thing added by toggle is two element attributes, `onLabel` and `offLabel`, and an override of `renderInput`.

Speaking of `renderInput`, I updated Checkbox to follow TextInput's model. The input element is now rendered in `renderInput`, which is called by `render`. Toggle overrides `renderInput` to add its own elements, then concats the output of Checkbox's `renderInput`.

I'm working on an update for @dojo/themes.

jason0x43 commented:
Hmmm...I'm not sure how theme inheritance works, but I may need to re-work it a bit.

tomdye commented:
Closing this as discussed with @jason0x43 as it has fallen stale due to RC changes and have raised issues around theme inheritance. @jason0x43 are you able to raise issues for what you discovered?
